pandas/tests/test_categorical.py

```diff
@@ -4100,6 +4100,16 @@ def f():
         res = df.fillna("a")
         tm.assert_frame_equal(res, df_exp)
 
+        # GH 14021
+        cat = Categorical([np.nan, 2, np.nan])
+        val = Categorical([np.nan, np.nan, np.nan])
```

Comment:
add a similar test that uses datetimelike and ``NaT`` (in fact ideally loop over DTI, TDI, PI)